### ISSUE TITLE
Link: Super-confident predictions 

### ISSUE THREAD

**Issue by tupini07** [Wikidata/soweego#350]
Implement the super confident predictions presented in #305 for the `link` module.

**Comment by tupini07** [comment]
The following table was generated by running the following command on all _catalogs_, _entites_, and _join methods_

``` fish
python -m soweego linker link all $catalog $entity -jm $join_method
```

**Comment by tupini07** [comment]
| Join Method  | Catalog     | Entity   | # of Predictions | Mean     | STD      |
|--------------|-------------|----------|------------------|----------|----------|
| Union        | IMDB        | Actor    | 50927            | 0.945392 | 0.147841 |
| Union        | IMDB        | Director | 4815             | 0.979298 | 0.073768 |
| Union        | IMDB        | Musician | 45036            | 0.985589 | 0.069565 |
| Union        | IMDB        | Producer | 1386             | 0.997396 | 0.028477 |
| Union        | IMDB        | Writer   | 6353             | 0.991922 | 0.054631 |
| Union        | Discogs     | Band     | 24058            | 0.981553 | 0.084890 |
| Union        | Discogs     | Musician | 45970            | 0.965588 | 0.115180 |
| Union        | Musicbrainz | Band     | 14777            | 0.914675 | 0.148736 |
| Union        | Musicbrainz | Musician | 30649            | 0.985456 | 0.072673 |
| Intersection | IMDB        | Actor    | 50927            | 0.945391 | 0.147841 |
| Intersection | IMDB        | Director | 4815             | 0.979297 | 0.073768 |
| Intersection | IMDB        | Musician | 45036            | 0.985589 | 0.069565 |
| Intersection | IMDB        | Producer | 1386             | 0.997395 | 0.028476 |
| Intersection | IMDB        | Writer   | 6353             | 0.991921 | 0.054630 |
| Intersection | Discogs     | Band     | 24058            | 0.981552 | 0.084890 |
| Intersection | Discogs     | Musician | 45970            | 0.965588 | 0.115180 |
| Intersection | Musicbrainz | Band     | 14777            | 0.914674 | 0.148736 |
| Intersection | Musicbrainz | Musician | 30649            | 0.985456 | 0.072673 |
| Average      | IMDB        | Actor    | 39910            | 0.782736 | 0.121685 |
| Average      | IMDB        | Director | 4425             | 0.921095 | 0.033104 |
| Average      | IMDB        | Musician | 42883            | 0.865742 | 0.080960 |
| Average      | IMDB        | Producer | 1369             | 0.859025 | 0.078662 |
| Average      | IMDB        | Writer   | 5914             | 0.751406 | 0.164815 |
| Average      | Discogs     | Band     | 22885            | 0.794083 | 0.118360 |
| Average      | Discogs     | Musician | 40309            | 0.808705 | 0.118197 |
| Average      | Musicbrainz | Band     | 8439             | 0.819419 | 0.119028 |
| Average      | Musicbrainz | Musician | 29093            | 0.855870 | 0.081999 |

**Comment by tupini07** [comment]
![Classification Counts](https://user-images.githubusercontent.com/3422347/63259869-d9124880-c27f-11e9-8689-4c5f5427b1c1.png)

**Comment by tupini07** [comment]
![Classification Mean](https://user-images.githubusercontent.com/3422347/63260193-900ec400-c280-11e9-8157-03a9fa8fcef8.png)

**Comment by tupini07** [comment]
![Figure_1](https://user-images.githubusercontent.com/3422347/63260310-d9f7aa00-c280-11e9-805e-9781cbe35d4b.png)

**Comment by tupini07** [comment]
![Figure_1](https://user-images.githubusercontent.com/3422347/63260385-188d6480-c281-11e9-90e8-b2a0d49a5e0e.png)

**Comment by tupini07** [comment]
Necessary changes have been made to the linker module and results have been posted to this issue.

The changes are in the `super-confident-predictions` branch, and will be merged into _master_ once #349 is complete.

**Comment by tupini07** [comment]
Reopening issue: 
In case of duplicate predictions when doing the _union_ or _intersection_ joins of the predictions we should use a _majority vote_ to decide if a prediction will end up in the final "linking" set or not.

Currently we take the "optimistic", or best, prediction as the final one and discard the duplicates

**Comment by tupini07** [comment]
As an idea: We could separate the _join method_ into (_union_ and _intersection_), and the _combination method_ into (_average_ or _majority vote_). And let the user decide which combination should be used

**Comment by tupini07** [comment]
Functionality has been implemented. Now only need to update the results of the table above (as well as the graphs)

**Comment by tupini07** [comment]
updates have been made. Marking this task as complete

**Comment by tupini07** [comment]
Reopening task since we need to reevaluate the performance after adding the 'random forest' classifier (see #355)
